core/unit_test/TestStackTrace.hpp

```diff
@@ -0,0 +1,180 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#include <iostream>
+#include <gtest/gtest.h>
+#include "Kokkos_Core.hpp"
+
+#include <impl/Kokkos_Stacktrace.hpp>
+
+namespace Test {
+
+void stacktrace_test_f0(std::ostream& out);
+
+int stacktrace_test_f1(std::ostream& out);
+
+void stacktrace_test_f2(std::ostream& out);
+
+int stacktrace_test_f3(std::ostream& out, const int level);
+
+void stacktrace_test_f4();
+
+void my_fancy_handler();
+
+int test_stacktrace(bool bTerminate) {
+  bool success = true;
+
+  stacktrace_test_f1(std::cout);
+
+  // TODO figure out whether stacktraces give function names at all (i.e. did we
+  // compile with -rdynamic)
+  {
+    std::stringstream sstream;
+    Kokkos::Impl::print_saved_stacktrace(sstream);
+
+    std::string f1output = sstream.str();
+    auto found_pos_f0    = f1output.find("stacktrace_test_f0");
+    bool not_found_f0    = (found_pos_f0 == std::string::npos);
+    auto found_pos_f1    = f1output.find("stacktrace_test_f1");
+    bool found_f1        = !(found_pos_f1 == std::string::npos);
+    auto found_pos_f2    = f1output.find("stacktrace_test_f2");
+    bool not_found_f2    = (found_pos_f2 == std::string::npos);
+    auto found_pos_f3    = f1output.find("stacktrace_test_f3");
+    bool not_found_f3    = (found_pos_f3 == std::string::npos);
+    auto found_pos_f4    = f1output.find("stacktrace_test_f4");
+    bool not_found_f4    = (found_pos_f4 == std::string::npos);
```

Comment:
```C++
for (auto x : {"foo", "bar"})
{
    ASSERT_TRUE(stacktrace.find(x) == std::string::npos);
}
```

Reply:
addressed in https://github.com/kokkos/kokkos/pull/2226/commits/fd4b9f45cf8d399b47d00a1cba7eea802697c0ff